File: datautils/src/main/java/dev/trindadedev/lib/ui/components/preference/Preference.java
package dev.trindadedev.lib.ui.components.preference;

import android.content.Context;
import android.content.res.TypedArray;
import android.util.AttributeSet;
import android.view.LayoutInflater;
import android.widget.LinearLayout;

import dev.trindadedev.lib.ui.components.R;
import dev.trindadedev.lib.ui.components.databinding.LayoutPreferenceBinding;

public class Preference extends LinearLayout  {
  private LayoutPreferenceBinding bind;
  

  public Preference(Context context) {
    this(context, null);
  }

  public Preference(Context context, AttributeSet attrs) {
    this(context, attrs, 0);
    init(context,attrs);
  }

  public Preference(Context context, AttributeSet attrs, int defStyleAttr) {
    super(context, attrs, defStyleAttr);
    init(context, attrs);
  }

  private void init(Context context, AttributeSet attrs) {
     bind = LayoutPreferenceBinding.inflate(LayoutInflater.from(context));

    TypedArray a = context.getTheme().obtainStyledAttributes(attrs, R.styleable.Preference, 0, 0);

    try {
      String title = a.getString(R.styleable.Preference_preferenceTitle);
      String description = a.getString(R.styleable.Preference_preferenceDescription);
      bind.preferenceTitle.setText(title != null ? title : "");
      bind.preferenceDescription.setText(description != null ? description : "");
    } finally {
      a.recycle();
    }
    if(bind != null) {
      removeAllViews();
    	addView(bind.getRoot());
    }
  }


  public void setTitle(String value) {
    bind.preferenceTitle.setText(value);
  }

  public void setDescription(String value) {
    bind.preferenceDescription.setText(value);
  }
  
  public void setOnClickListener(OnClickListener c){
    bind.getRoot().setOnClickListener(c);
  }
  
}

File: datautils/src/main/java/dev/trindadedev/lib/ui/components/preference/withicon/Preference.kt
package dev.trindadedev.lib.ui.components.preference.withicon

import android.content.Context
import android.util.AttributeSet
import android.view.LayoutInflater
import android.view.View
import android.widget.ImageView
import android.widget.LinearLayout
import android.widget.TextView

import androidx.annotation.DrawableRes

import dev.trindadedev.lib.ui.components.R

class Preference @JvmOverloads constructor(
    context: Context,
    attrs: AttributeSet? = null,
    defStyleAttr: Int = 0
) : LinearLayout(context, attrs, defStyleAttr) {

    public val preferenceTitle: TextView
    public val preferenceDescription: TextView
    public val preferenceIcon: ImageView
    public val preference: View

    init {
        LayoutInflater.from(context).inflate(R.layout.layout_preference_withicon, this, true)

        preferenceTitle = findViewById(R.id.preference_title)
        preferenceDescription = findViewById(R.id.preference_description)
        preferenceIcon = findViewById(R.id.preference_icon)
        preference = findViewById(R.id.preference)

        context.theme.obtainStyledAttributes(
            attrs,
            R.styleable.Preference,
            0, 0
        ).apply {
            try {
                val title = getString(R.styleable.Preference_preferenceTitle) ?: ""
                val description = getString(R.styleable.Preference_preferenceDescription) ?: ""
                val iconResId = getResourceId(R.styleable.Preference_preferenceIcon, 0)

                preferenceTitle.text = title
                preferenceDescription.text = description

                if (iconResId != 0) {
                    preferenceIcon.setImageResource(iconResId)
                } else {
                    preferenceIcon.visibility = View.GONE
                }
            } finally {
                recycle()
            }
        }
    }

    fun setPreferenceClickListener(listenerClick: View.OnClickListener) {
        preference.setOnClickListener(listenerClick)
    }

    fun setTitle(value: String) {
        preferenceTitle.text = value
    }

    fun setDescription(value: String) {
        preferenceDescription.text = value
    }

    fun setIcon(@DrawableRes resId: Int) {
        if (resId != 0) {
            preferenceIcon.setImageResource(resId)
            preferenceIcon.visibility = View.VISIBLE
        } else {
            preferenceIcon.visibility = View.GONE
        }
    }
}